### DISCUSSION TITLE
not able to upload file successfully over http using ngrok/over LAN

### DISCUSSION THREAD

**Discussion by krishnardt** [fastapi/fastapi#6831] [Questions]
I tried to upload a file from localhost in the same system. It is working fine.
When I tried to upload file from other other device over LAN, it is just creating a file but not writing the data into file object.
I also tried ngrok to upload file from a device outside of LAN, it is just creating a file but not writing the data into file object.
all the trails I did till now are over http
 

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```
from fastapi.middleware.cors import CORSMiddleware
from fastapi import Request, Depends
from fastapi import FastAPI
import uvicorn


app = FastAPI(title='xxxxxxxxxxx',
        description='xxxxxxxxxxxxxxxxx API',
        version='1.0.0', redoc_url = None,)

origins = [
    "*",
]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

@app.post("/uploadfiles/")
async def create_upload_files(files: List[UploadFile] = File(...)):#, file_size: int = Depends(valid_content_length)):

    allowed_file_content = ['application/zip', 'image/jpeg', 'image/jpg']
    dest_path = "E://************************/"
  

    for file in files:
        async with aiofiles.open(os.path.join(dest_path, file.filename), 'wb') as folder:
            content = await file.read()
            await folder.write(content)



@app.get("/")
async def main():
    content = """
<body>
<form action="/uploadfiles/" enctype="multipart/form-data" method="post">
<input name="files" type="file" multiple>
<input type="submit">
</form>
</body>
    """
    return HTMLResponse(content=content)

if __name__ == "__main__":
    uvicorn.run("pilot:app", host="0.0.0.0", port=8000, reload=True)

```

### Description

* Open the browser(from different device or use ngrok) and call the endpoint `/`.
* It returns a JSON with html file`.
* But I expected it to return the full file in the destination path.

### Environment

* OS: [Windows]:
* FastAPI Version [0.63.0]:

### Additional context

Headers({'host': '192.168.1.30:8000', 'connection': 'keep-alive', 'content-length': '339374', 'cache-control': 'max-age=0', 'upgrade-insecure-requests': '1', 'origin': 'http://192.168.1.30:8000', 'content-type': 'multipart/form-data; boundary=----WebKitFormBoundaryu3nR8DFKxTkBxBH8', 'user-agent': 'Mozilla/5.0 (Linux; Android 9; VTR-L29) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.93 Mobile Safari/537.36', 'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9', 'referer': 'http://192.168.1.30:8000/upload_file', 'accept-encoding': 'gzip, deflate', 'accept-language': 'en-US,en;q=0.9'}) 

followed this reference too.. but no use.
[https://stackoverflow.com/questions/63580229/how-to-save-uploadfile-in-fastapi](url)

**Comment by krishnardt** [comment]
Hi thanks, the issue is resolved.it is due to firewall.

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
